README.md

```diff
@@ -315,17 +307,30 @@ that it can be deployed in. There are generally two ways to get trace data
 out of the system.
 
 The first is to use an I/O interface on your device and use the `report` API to
-send data to a waiting collector, like in this UDP oriented example below:
-
-```rust
-fn send_report<'a>(socket: &UdpSocket, probe: &mut ModalityProbe<'a>, report_buffer: &mut [u8]) {
-    let n_report_bytes = probe
-        .report(report_buffer)
-        .expect("Could not produce a report");
-    if let Some(nonzero_report_size) = n_report_bytes {
-        socket
-            .send_to(&report_buffer[..nonzero_report_size.get()], COLLECTOR_ADDR)
-            .expect("Could not send_to");
+send data to a waiting collector, like in this UDP-oriented example below:
+
+```c
+static void send_report(modality_probe * const probe)
+{
+  {
```

Comment:
Looks like you've still got an extra set of `{..}` brackets here. Check out the snippet on master in the modality-probe-capi/README.md for syntax.

Reply:
```c
static void send_report(modality_probe * const probe)
{
    size_t report_size;
    const size_t err = modality_probe_report(
            probe,
            &g_report_buffer[0],
            sizeof(g_report_buffer),
            &report_size);
    assert(err == MODALITY_PROBE_ERROR_OK);

    if(report_size != 0)
    {
        const ssize_t status = sendto(
                g_report_socket,
                &g_report_buffer[0],
                report_size,
                0,
                (const struct sockaddr*) &g_collector_addr,
                sizeof(g_collector_addr));
        assert(status != -1);
    }
}
```

Reply:
Thanks for the catch!

Reply:
:+1: Change made